azurelinuxagent/ga/update.py

```diff
@@ -105,9 +105,10 @@ def __init__(self, vm_status=None):
             self.summary = []
             self.converged = True
         else:
-            self.summary = [(h.extension_status.name, h.extension_status.status) for h in vm_status.vmAgent.extensionHandlers]
+            # take the name and status of the extension if is it not None, else use the handler's
+            self.summary = [(o.name, o.status) for o in map(lambda h: h.extension_status if h.extension_status is not None else h, vm_status.vmAgent.extensionHandlers)]
             self.summary.sort(key=lambda s: s[0])  # sort by extension name to make comparisons easier
-            self.converged = all(status in (ValidHandlerStatus.success, ValidHandlerStatus.error) for _, status in self.summary)
+            self.converged = all(status in (ValidHandlerStatus.success, ValidHandlerStatus.error, HandlerStatus.ready) for _, status in self.summary)
```

Comment:
When looking at HandlerStatus, both Ready and NotReady imply that the Handler is `Converged`. They're just used to identify if the handler failed or not, but it still means that it converged. You should add `HandlerStatus.not_ready` here too

Reply:
yes, good point